crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -321,10 +322,17 @@ where
         }
     }
 
-    let reconstructed_contents = reconstructor
-        .add_segment(segment_pieces.as_ref())
-        .map_err(|error| error.to_string())?;
-    drop(segment_pieces);
+    // CPU-intensive code can block the executor, and the reconstructor reference can't escape this function.
+    let mut reconstructor_tmp = reconstructor.clone();
+    let (reconstructor_tmp, reconstructed_contents) = spawn_blocking(move || {
+        let contents = reconstructor_tmp.add_segment(segment_pieces.as_ref());
+        drop(segment_pieces);
+        contents.map(|contents| (reconstructor_tmp, contents))
+    })
+    .await
+    .expect("Panic if blocking task panicked")
+    .map_err(|error| error.to_string())?;
+    *reconstructor = reconstructor_tmp;
```

Comment:
I'd wrap it in a `Arc<Mutex<>>` instead in this case so you can move a cheap clone into a task and access inner contents that way directly without doing awkward clone just to throw it away afterwards.

Also dropping `segment_pieces` is unnecessary since it is already moved into a task that is about to exit (it was not needed before either, likely left after some refactoring).

Reply:
Done! This meant modifying the callers, because an `Arc<Mutex<&mut Reconstructor>>` still has a limited lifetime.